### PR TITLE
RHDEVDOCS-3570 Docs: Use sidecar pattern for Jenkins pod templates

### PR DESCRIPTION
**Pull request by rolfedh**
- Aligned team: Dev Tools
- For branches: 4.10+
- Jira: https://issues.redhat.com/browse/RHDEVDOCS-3570
- Direct link to doc preview: 
  - https://deploy-preview-39873--osdocs.netlify.app/openshift-enterprise/latest/openshift_images/using_images/images-other-jenkins
  - https://deploy-preview-39873--osdocs.netlify.app/openshift-enterprise/latest/openshift_images/using_images/images-other-jenkins-agent
- SME review: reviewed and approved: @gabemontero 
- QE review: pending completion of feature by @akram: @prietyc123
- Peer review: @rolfedh 
- <All reviews complete. Please merge now>

### CODE REVIEW

**Review by gabemontero — CHANGES_REQUESTED** [review]
a minor clarification/addition the text I provided to you previously @rolfedh

**Review by gabemontero — APPROVED** [review]
/approve
/lgtm

File: modules/images-other-jenkins-config-kubernetes.adoc [hunk 1]
```diff
@@ -76,11 +76,72 @@ data:
     </org.csanchez.jenkins.plugins.kubernetes.PodTemplate>
 ----
 
+The following example shows two containers that reference image streams that are running in the `openshift` namespace. One container handles the JNLP contract for launching Pods as Jenkins Agents. The other container uses an image with tools for building code in a particular coding language:
```
Comment (gabemontero) [inline modules/images-other-jenkins-config-kubernetes.adoc:79]:
```suggestion
The following example shows two containers that reference image streams that are present in the `openshift` namespace. One container handles the JNLP contract for launching Pods as Jenkins Agents. The other container uses an image with tools for building code in a particular coding language:
```

File: modules/images-other-jenkins-config-kubernetes.adoc [hunk 2]
```diff
@@ -76,11 +76,72 @@ data:
...
+
+[source,yaml]
+----
+kind: ConfigMap
+apiVersion: v1
+metadata:
+  name: jenkins-agent
+  labels:
+    role: jenkins-slave
```
Comment (prietyc123) [inline modules/images-other-jenkins-config-kubernetes.adoc:88]: I am bit confused here with labels `role: jenkins-slave`. AFAIK `jenkins-slave` is legacy one and IIUC it should be `jenkins-agent`, no? 

@akram please correct me if I am wrong.

**Review by prietyc123 — CHANGES_REQUESTED** [review]

File: modules/images-other-jenkins-config-kubernetes.adoc (hunk 2 above)
Comment (akram) [reply modules/images-other-jenkins-config-kubernetes.adoc:88]: @prietyc123 you are correct. We should use the `jenkins-agent` terminology now even though internally, the legacy label is still handled.

Comment (rolfedh) [reply modules/images-other-jenkins-config-kubernetes.adoc]: Updated. Thanks. I've also verified that there are no other instances of "jenkins slave" or "jenkins-slave" in the documentation.

**Comment by gabemontero** [comment]
@prietyc123 @rolfedh @akram let's be sure that using both labels in the new examples provided work when @prietyc123 verifies both JNKS-257 and https://bugzilla.redhat.com/show_bug.cgi?id=2034839

**Comment by gabemontero** [comment]
Let's go ahead and merge this @rolfedh if you are ready ....any tweaks needed based on @prietyc123 's testing can come under new PRs under https://bugzilla.redhat.com/show_bug.cgi?id=2034839 or any new BZ's she opens.

**Comment by rolfedh** [comment]
Thanks, @gabemontero. @prietyc123 and @akram, if you both agree, please sign off on this PR with `/lgtm` and I'll get it merged.

**Comment by prietyc123** [comment]
lgtm

**Comment by gabemontero** [comment]
@rolfedh - during are grooming today we discovered that one of the doc items we need for this story was declaring that the existing PodTemplates are deprecated (the non side car pod templates)

I don't think we did that in this PR.

We'll also want a deprecation item in the 4.10 release notes.

How should we go about tracking that additional work?  Bugzilla ? Jira work item?

@adambkaplan FYI

**Comment by rolfedh** [comment]
> @rolfedh - during are grooming today we discovered that one of the doc items we need for this story was declaring that the existing PodTemplates are deprecated (the non side car pod templates)
> 
> I don't think we did that in this PR.
> 
> We'll also want a deprecation item in the 4.10 release notes.
> 
> How should we go about tracking that additional work? Bugzilla ? Jira work item?
> 
> @adambkaplan FYI

Thanks for catching that, @gabemontero. I've created https://issues.redhat.com/browse/RHDEVDOCS-3644 to track that work.

**Pull request merged**